src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialLocalizationSession.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.SpatialAlignment;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    public abstract class SpatialLocalizationSession : DisposableBase, ISpatialLocalizationSession
+    {
+        protected readonly CancellationTokenSource defaultCTS = null;
+
+        /// <inheritdoc />
+        public abstract IPeerConnection Peer { get; }
+
+        /// <inheritdoc />
+        public SpatialLocalizationSession()
+        {
+            this.defaultCTS = new CancellationTokenSource();
+        }
+
+        /// <inheritdoc />
+        protected override void OnManagedDispose()
+        {
+            defaultCTS.Dispose();
+        }
+
+        /// <inheritdoc />
+        public abstract Task<ISpatialCoordinate> LocalizeAsync(CancellationToken cancellationToken);
+
+        /// <inheritdoc />
+        public void Cancel()
+        {
+            if (!this.IsDisposed &&
+                defaultCTS != null &&
+                defaultCTS.Token.CanBeCanceled)
+            {
+                defaultCTS.Cancel();
+            }
+        }
+
+        /// <inheritdoc />
+        public abstract void OnDataReceived(BinaryReader reader);
```

Comment:
>public abstract void OnDataReceived(BinaryReader reader); [](start = 8, length = 57)

I would just make this virtual with an empty body, since it's sometimes not needed by subclasses